### PR TITLE
Adding changelog for v0.10.1

### PR DESCRIPTION
**Pull request by pokearu**
*Description of changes:*
Adding changelog for release `v0.10.1`

*Testing:*
`make serve -C ./docs`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/reference/changelog.md [hunk 1]
```diff
@@ -6,6 +6,15 @@ weight: 35
 
 ## Unreleased
 
+## [v0.10.1](https://github.com/aws/eks-anywhere/releases/tag/v0.10.1)
+
+### Changed
+- Updated EKS Distro versions to latest release
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/changelog.md:12]: Do we need to update the [Artifacts ](https://anywhere.eks.amazonaws.com/docs/reference/artifacts/) in the docs to reflect new Ubuntu and Bottlerocket OS images for the release?

**Review by chrisnegus — COMMENTED** [review]
Just one question about whether we need to update the artifacts docs.

File: docs/content/en/docs/reference/changelog.md (hunk 1 above)
Comment (pokearu) [reply docs/content/en/docs/reference/changelog.md:12]: Yes! will do in a follow up.

File: docs/content/en/docs/reference/changelog.md [hunk 2]
```diff
@@ -6,6 +6,15 @@ weight: 35
...
+
+Fixed
```
Comment (vivek-koppuru) [inline docs/content/en/docs/reference/changelog.md:14]:
```suggestion
### Fixed
```

**Review by mitalipaygude — APPROVED** [review]

**Review by vivek-koppuru — APPROVED** [review]

**Pull request merged**